src/shared/interfaces/categorized-validation-errors.ts

```diff
@@ -0,0 +1,6 @@
+import { SchemaValidationErrors } from './schema-validation-errors';
+
+export interface CategorizedValidationErrors {
+  Errors: SchemaValidationErrors;
+  Warnings?: SchemaValidationErrors;
```

Comment:
Because right now on internalErrors we have only errors so I make it optional to avoid every time initializing `Warnings` object just for the types. Also, given the name `CategorizedValidationErrors` someone expects to find `Errors` and `Warnings` may be optional.